### PR TITLE
add methods to expose n to pySearcher

### PR DESCRIPTION
**Pull request by stephaniewhoo**
I think leaving extracting the number here would be easier since we don't need to import IndexReader and IndexSearcher to the pyserini.

### CODE REVIEW

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 1]
```diff
@@ -206,6 +206,17 @@ public void setBM25Similarity(float k1, float b) {
     searcher.setSimilarity(similarity);
   }
 
+  /**
+   * Returns the number of documents in the provided index file. Used to expose the total number of documents in Pyserini
+   *
+   * @return the number of documents in total
+   */
+
+   public Int getTotalNumDocuments(){
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleSearcher.java:208]: primitive int instead if object Int.
I think there's already an `IndexSearcher` for you to read from? see line 96.

Comment (stephaniewhoo) [reply src/main/java/io/anserini/search/SimpleSearcher.java:208]: I found line 96 was initialized to be a null and was assigned by some values in `setLMDirichletSimilarity`. Not sure if there are other assignments for this searcher

Comment (lintool) [reply src/main/java/io/anserini/search/SimpleSearcher.java:208]: IIRC, the code initializes an `IndexSearcher` on demand, as in: https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/SimpleSearcher.java#L308

You method should probably do the same also?

If there are any cases that would yield an NPE, can you please add a test case to catch, and then fix in code while you're at it?

Comment (stephaniewhoo) [reply src/main/java/io/anserini/search/SimpleSearcher.java]: added test cases, still trying to catch possible NPE. For now, haven't encountered that.

File: src/main/java/io/anserini/search/SimpleSearcher.java [hunk 2]
```diff
@@ -199,6 +199,19 @@ public void setBM25Similarity(float k1, float b) {
...
+  /**
+   * Returns the number of documents in the provided index file. Used to expose the total number of documents in Pyserini
+   *
+   * @return the number of documents in total
+   */
+
+   public int getTotalNumDocuments(){
+     if (searcher == null) {
+      searcher = new IndexSearcher(reader);
+    }
```
Comment (lintool) [inline src/main/java/io/anserini/search/SimpleSearcher.java:211]: fix indentation please.

File: src/test/java/io/anserini/search/SimpleSearcherTest.java [hunk 3]
```diff
@@ -249,4 +249,11 @@ public void testFieldedBatchSearch() throws Exception {
 
     searcher.close();
   }
+
+  @Test
+  public void testTotalNumDocuments() throws Exception {
+    SimpleSearcher searcher = new SimpleSearcher(super.tempDir1.toString());
+    assertEquals(3,searcher.getTotalNumDocuments());
```
Comment (lintool) [inline src/test/java/io/anserini/search/SimpleSearcherTest.java:256]: space after 3, 
remove empty newline.

**Comment by lintool** [comment]
@stephaniewhoo can you please address those minor changes?

**Comment by stephaniewhoo** [comment]
> @stephaniewhoo can you please address those minor changes?

Done formatting. Will be careful about the format next time! Sorry

**Pull request merged**